### PR TITLE
feat: enable IWDG in all applications

### PR DESCRIPTION
**Pull request by amitlissack**
This PR enables the IWDG. It expires after 1 second. When the watchdog expires the system will be reset.

A task is introduced that will refresh the watchdog timer every 950ms.

The bootloader detects when a restart occurred because of IWDG and does not automatically jump to the main application.

### CODE REVIEW

File: common/firmware/system/iwdg.c [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+#include "common/firmware/iwdg.h"
+
+
+static IWDG_HandleTypeDef hiwdg;
+
+
+void MX_IWDG_Init(void) {
+    /* USER CODE END IWDG_Init 1 */
+    hiwdg.Instance = IWDG;
+    hiwdg.Init.Prescaler = IWDG_PRESCALER_32;
```
Comment (amitlissack) [inline common/firmware/system/iwdg.c:12]: 32khz clock prescaled by 32 (1khz clock).

**Comment by fsinapi** [comment]
Have you tested how the watchdog behaves when you hit a breakpoint in the debugger? Not sure if you need to set the `DBGMCU` registers to pause the timer during breakpoints

**Comment by amitlissack** [comment]
> Have you tested how the watchdog behaves when you hit a breakpoint in the debugger? Not sure if you need to set the `DBGMCU` registers to pause the timer during breakpoints

I have. Sitting at a breakpoint doesn't trigger a watchdog reset.

**Review by sfoster1 — APPROVED** [review]
Nice!

I'm interested to see how this approach with a dedicated, prioritized task goes going forward - I'm more used to using the idle task for this. I think it'll be fine, just not quite what I'm used to.

**Pull request merged**